my-react-app/src/components/Form/Form.tsx

```diff
@@ -0,0 +1,174 @@
+import PhoneInput from "react-phone-number-input/input";
+import { useForm } from "react-hook-form";
+import "react-phone-number-input/style.css";
+import CountryList from "./CountryList";
+import { useState } from "react";
+
+interface FormData {
+  firstName: string;
+  lastName: string;
+  title?: string;
+  street?: string;
+  city?: string;
+  zip: string;
+  country: string;
+  phoneNumber?: string;
+}
+
+const Form = () => {
+  const [check, setCheck] = useState(false);
+  /*  const SubmitTrue = () => {
+    return check ? (
+      <p className="bg-green-300 p-3 text-white rounded-lg m-auto">
+        Submit successful
+      </p>
+    ) : null;
+  }; */
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    getValues,
+  } = useForm();
+
+  const onSubmit = (data: FormData) => {
+    console.log(data);
+    setCheck(true);
+  };
+
+  return (
+    <form
+      className="border-2 rounded-lg p-3 flex flex-row flex-wrap bg-slate-50"
+      /*onSubmit={(e: FormEvent<className="my-2 text-xl font-bold" htmlFormElement>) => {
+          e.preventDefault(); // Verhindert das Standardverhalten des Formulars
+          const formData = new FormData(e.target as className="my-2 text-xl font-bold" htmlFormElement); // Extrahiere Formulardaten
+          const data: FormData = {
+            firstName: formData.get("firstName") as string,
+            lastName: formData.get("lastName") as string,
+            title: formData.get("title") as string,
+            street: formData.get("street") as string,
+            city: formData.get("city") as string,
+            zip: formData.get("zip") as string,
+            country: formData.get("country") as string,
+            phoneNumber: formData.get("phoneNumber") as string,
+          };
+          console.log(data);
+        }}*/
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      <div className="form-group m-5 p-3 flex flex-col items-start basis-full">
+        <label className="my-2 text-xl font-bold" htmlFor="firstName">
+          First Name *
+        </label>
+        <input
+          placeholder="Your first name"
+          className="px-2 w-full border-2 rounded-lg  hover:shadow-lg focus:outline-none focus:ring focus:ring-violet-300 min-h-12 text-violet-300"
```

Comment:
sehr schön design nur haben wir jetzt immer wieder diese lange schlange an classes.. das könntest du lösen dass du das textinput componente in eine extrige componennte auslagerst (mit eigenem file) wo du nur mehr die notwendigen props hineingibst .. das kannst du dir zb im sitelifecode oder im internet anschauen wie man so etwas macht